### PR TITLE
F/app serializers

### LINKED ISSUE #53: Work out a solution for people to be able to build/use a type of AppSerializer Class
In the same way that AppControllers, etc operate.

### PR DESCRIPTION
**Pull request by jtyost2**
Adds an AppSerializer class and documentation regarding the use of it.

Note there are no breaking changes in this PR, however the documentation is updated to reflect that AppSerializer class will probably be the primary base Serializer object developers will build upon.
Closes #53

### CODE REVIEW

**Comment by jtyost2** [comment]
Assigning to @beporter for review and feedback.

**Comment by beporter** [comment]
One specific thing comes to mind:

Does this work even if the consuming  Cake project does not have a `/Serializer/AppSerializer.php` file defined? I'm thinking of how Cake controller's can `extend AppController` without there actually needed to be a `/Controller/AppController.php` file in your project. 

It seems like the `App` class needs to know about `Plugin/Serializers/Serializer/` as a load path for Serializer objects. All a `/Serializer/SomethingSerializer.php` file should need to do is `App::uses('AppSerializer', 'Serializer');` and have it work whether or not `/Serializer/AppSerializer.php` exists by falling back to `Plugin/Serializers/Serializer/AppSerializer.php`.

In other words: It should work exactly like AppController and AppModel.

**Comment by jtyost2** [comment]
@beporter Throwing back for re-review.

**Comment by beporter** [comment]
Nice work, this looks like it mirrors the behavior of Cake's core classes (Models, Controllers, Components, etc.) when it comes to using a core `AppWhatever` when one is not present in the app.

:+1:

**Comment by jtyost2** [comment]
Need to run through this PR and verify it internally before merging.
